lib/src/chatbox.dart

```diff
@@ -237,15 +239,27 @@ class ChatBoxState extends State<ChatBox> {
       },
       shouldOverrideUrlLoading: (InAppWebViewController controller, NavigationAction navigationAction) async {
         if (navigationAction.navigationType == NavigationType.LINK_ACTIVATED) {
-          if (await launchUrl(navigationAction.request.url!)) {
-            // We launched the browser, so we don't navigate to the URL in the WebView
-            return NavigationActionPolicy.CANCEL;
-          } else {
-            // We couldn't launch the external browser, so as a fallback we're using the default action
-            return NavigationActionPolicy.ALLOW;
+          var shouldOpenBrowser = true;
+
+          if (widget.onUrlNavigation != null) {
+            // The onUrlNavigation function has been defined, so let's see if we should open the browser or not
+            if (widget.onUrlNavigation!(UrlNavigationRequest(navigationAction.request.url!.rawValue)) == UrlNavigationAction.deny) {
+              shouldOpenBrowser = false;
+            }
+          }
+
+          if (shouldOpenBrowser) {
+            if (await launchUrl(navigationAction.request.url!)) {
+              // We launched the browser, so we don't navigate to the URL in the WebView
+              return NavigationActionPolicy.CANCEL;
+            } else {
+              // We couldn't launch the external browser, so as a fallback we're using the default action
+              return NavigationActionPolicy.ALLOW;
+            }
           }
         }
-        return NavigationActionPolicy.ALLOW;
+
+        return NavigationActionPolicy.CANCEL;
```

Comment:
What's the rational behind changing this default return from `NavigationActionPolicy.ALLOW`?

Reply:
I complicated the logic a little too much, my bad